internal/mage/go.mod

```diff
@@ -20,10 +21,9 @@ require (
 	github.com/mitchellh/go-homedir v1.1.0 // indirect
 	github.com/pkg/errors v0.9.1 // indirect
 	github.com/vektah/gqlparser/v2 v2.5.6 // indirect
-	golang.org/x/sys v0.13.0 // indirect
+	golang.org/x/sys v0.14.0 // indirect
 )
 
-// needed to resolve "ambiguous import: found package cloud.google.com/go/compute/metadata in multiple modules"
-replace cloud.google.com/go => cloud.google.com/go v0.100.2
-
+// NOTE: this should be removed once internal/distconsts/ package makes it into
+// the next release
```

Comment:
If we remove this, then in a scenario where we want to change one of the paths encoded in the `distconsts` package (i.e. change the path of the shim binary), then we'd update it in `github.com/dagger/dagger` but it would not be picked up by `github.com/dagger/dagger/internal/mage`, right?

So then wouldn't that leave us in a state where we can't actually build the dev engine until the dev engine has been released (catch-22)?

Or am I misthinking somewhere?

Reply:
> I think I was imagining another layer: building the dev engine in the dev engine. But I don't think we do this?

We do actually do that. We added it to catch obscure (but really awful) issues that popped up occasionally where the dev engine accidentally ended up incompatible w/ itself for various reasons.

[This `testdev` workflow does it](https://github.com/sipsma/dagger/blob/9a0f81a7367a675b13854f0be82694d4ebc44dd3/.github/workflows/engine-and-cli.yml#L36-L36), specifically by triggering [this yaml-bash codepath](https://github.com/sipsma/dagger/blob/9a0f81a7367a675b13854f0be82694d4ebc44dd3/.github/workflows/_hack_make.yml#L58-L58) (😞)

What's the specific issue you're thinking of that would relate back to the replace statement? I'm not immediately seeing the connection

Reply:
Ok, sorry, this is actually fine (I wrote several responses here arguing the other side, and deleted them since they were wrong).

When mage is running dagger instances, then it's only ever running the "dev" version that the current directory defines. Because of that, the `replace` directive is always valid (and required to stay valid).